frontend/dev/script.ts

```diff
@@ -0,0 +1,223 @@
+
+  const title = document.createElement('span');
+  title.classList.add('card__title');
+  title.textContent = item.name;
+
+  const wrapper = document.createElement('div');
+  wrapper.classList.add('wrapper');
+
+  const img = new Image();
+
+  img.onload = function () {
+    img.classList.add('card__img');
+    img.alt = ('Image of ' + item.name);
+
+    const info = document.createElement('div');
+    info.classList.add('wrapper', 'info');
+
+    const descrip = document.createElement('p');
+    descrip.classList.add('card__description');
+    descrip.textContent = item.descr.split(". ")[0] + ".";
+
+    const price = document.createElement('span');
+    price.classList.add('card__price');
+    price.textContent = item.price + "UAH.";
+
+    info.append(descrip, price);
+    wrapper.append(img, info);
+    li.append(title, wrapper);
+
+    if (itemsContainer) {
+      itemsContainer.appendChild(li);
+    }
+
+    if (isLast) {
+      hideSpinner();
+      showCatalog();
+      showPagination();
+    }
+  }
+
+  img.src = item.url;
+}
+
+function showSpinner(): void {
+  if (contentContainer) {
+    contentContainer.insertAdjacentHTML('afterbegin', `<i class="fas fa-compact-disc spinner" id="spinner"></i>`);
+  }
+}
+
+function hideSpinner(): void {
+  const spinner: HTMLElement | null = document.querySelector('#spinner');
+
+  if (spinner) {
+    spinner.style.display = 'none';
+  }
+}
+
+function showCatalog(): void {
+  if (itemsContainer) {
+    itemsContainer.style.display = 'flex';
+  }
+}
+
+function hideCatalog(): void {
+  if (itemsContainer) {
+    itemsContainer.style.display = 'none';
+  }
+}
+
+function showPagination(): void {
+  if (pagination) {
+    pagination.style.display = 'flex';
+  }
+}
+
+function hidePagination(): void {
+  if (pagination) {
+    pagination.style.display = 'none';
+  }
+}
+
+function nextPage(): void {
+  if (currentList.length > 0) {
+    currentPage++;
+    getDataAndUpdatePage();
+    updatePaginationButtons();
+  }
+}
+
+function prevPage(): void {
+  if (currentPage > 1) {
+    currentPage--;
+    getDataAndUpdatePage();
+    updatePaginationButtons();
+  }
+}
+
+function updatePageNum(): void {
+  if (currentPageSpan) {
+    currentPageSpan.textContent = String(currentPage);
+  }
+}
+
+async function updatePaginationButtons(): Promise<void> {
+  if (!buttonNext || !buttonPrev) {
+    return;
+  }
+  
+  if (await nextPageExists(currentPage + 1)) {
+    buttonNext.disabled = false;
+  } else {
+    buttonNext.disabled = true;
+  }
+
+  if (currentPage > 1) {
+    buttonPrev.disabled = false;
+  } else {
+    buttonPrev.disabled = true;
+  }
+}
+
+async function nextPageExists(page: number): Promise<boolean> {
+  try {
+    return (await fetch(`http://localhost:5500/api/sneakers/${page}`))
+      .json()
+      .then(list => {
+        if (list.length === 0) {
+          return false;
+        } else {
+          return true;
+        }
+      })
+  } catch (error) {
+    return false;
+  }
```

Comment:
maybe it's worth making it easier?) if we work with pagination, the backend should immediately return data on the number of pages